### PR TITLE
Use orientationIndex enum from header

### PR DESCRIPTION
**Pull request by cwoffenden**
As suggested in[ #21428](https://github.com/emscripten-core/emscripten/pull/21428#discussion_r1516423357) the enum value was taken from the header.

In the process, since adding to `struct_info.json` was needed, a fix to the reported path prompting to run `gen_struct_info.py` was added.

### CODE REVIEW

**Comment by cwoffenden** [comment]
Failures seem to be down to changes in `generated_struct_info64.json` not matching the older version.

File: src/generated_struct_info32.json [hunk 1]
```diff
@@ -185,6 +185,7 @@
         "EMSCRIPTEN_FULLSCREEN_SCALE_CENTER": 3,
         "EMSCRIPTEN_FULLSCREEN_SCALE_DEFAULT": 0,
         "EMSCRIPTEN_FULLSCREEN_SCALE_STRETCH": 1,
+        "EMSCRIPTEN_ORIENTATION_UNSUPPORTED": 0,
```
Comment (sbc100) [inline src/generated_struct_info32.json:188]: I believe the 64-bit version of this file needs updating too (
Run `tools/maint/gen_struct_info.py --wasm64`)

File: src/struct_info.json [hunk 2]
```diff
@@ -785,7 +785,13 @@
...
             "EMSCRIPTEN_DEVICE_MOTION_EVENT_SUPPORTS_ACCELERATION",
             "EMSCRIPTEN_DEVICE_MOTION_EVENT_SUPPORTS_ACCELERATION_INCLUDING_GRAVITY",
-            "EMSCRIPTEN_DEVICE_MOTION_EVENT_SUPPORTS_ROTATION_RATE"
+            "EMSCRIPTEN_DEVICE_MOTION_EVENT_SUPPORTS_ROTATION_RATE",
+
+            "EMSCRIPTEN_ORIENTATION_UNSUPPORTED",
+            "EMSCRIPTEN_ORIENTATION_PORTRAIT_PRIMARY",
+            "EMSCRIPTEN_ORIENTATION_PORTRAIT_SECONDARY",
+            "EMSCRIPTEN_ORIENTATION_LANDSCAPE_PRIMARY",
+            "EMSCRIPTEN_ORIENTATION_LANDSCAPE_SECONDARY"
```
[the remark this answers is not in the document]
Comment (cwoffenden) [reply src/struct_info.json]: I cut down to just add the single enum and run the script to update the generated files. I still get the error:
```
AssertionError: Unexpected difference:
--- /home/circleci/project/src/generated_struct_info64.json
+++ /home/circleci/project/src/generated_struct_info64.json.new
@@ -185,6 +185,7 @@
         "EMSCRIPTEN_FULLSCREEN_SCALE_CENTER": 3,
         "EMSCRIPTEN_FULLSCREEN_SCALE_DEFAULT": 0,
         "EMSCRIPTEN_FULLSCREEN_SCALE_STRETCH": 1,
+        "EMSCRIPTEN_ORIENTATION_UNSUPPORTED": 0,
         "EMSCRIPTEN_RESULT_DEFERRED": 1,
         "EMSCRIPTEN_RESULT_FAILED": -6,
         "EMSCRIPTEN_RESULT_FAILED_NOT_DEFERRED": -2,
```
Then I spent more than a few seconds to notice the file is `generated_struct_info64.json` but only the 32-bit generated file gets updated. I added it by hand, just to check that this is the issue.

**Review by sbc100 — APPROVED** [review]

**Comment by cwoffenden** [comment]
Note to a future me, because this will trip me up again in six months...
```
 emcc --clear-cache
 ./tools/maint/gen_struct_info.py
 ./tools/maint/gen_struct_info.py --wasm64
 ./tools/maint/gen_sig_info.py
```

**Pull request merged**